en/docs/apis/advanced-topics/connect-to-an-external-identity-provider.md

```diff
@@ -0,0 +1,146 @@
+    <td>The URL that returns the JSON web key (JWK) set of the IdP. This returns a collection of JSON Web Keys used to verify the signature of the JWT tokens.
+    </td>
+    </tr>
+    </table>
+
+     ![Identity provider details](../../assets/img/apis/advanced-topics/external-identity-providers/identity-provider-details.png){.cInlineImage-half}
+
+8.  Click **Add**. This adds Okta IdP to the list of existing identity providers in your organization.
+
+Now that you have added an external IdP, you can go ahead and obtain access tokens from that IdP to exchange and get Choreo access tokens to invoke APIs.
+
+
+## Obtain an access token from an external identity provider
+
+Follow this procedure to obtain an access token from an external IdP and exchange it with a Choreo token within your organization:
+
+1. Sign in to the Choreo API Developer Portal at [https://devportal.choreo.dev/](https://devportal.choreo.dev/).
+
+2. Click the **Applications** tab.
+
+3. Create a new application or use an existing application.
+
+4. In the left navigation menu, click **OAuth2 Tokens** under **Production Keys**. This displays the following page if you have not generated the **Consumer Key** and **Consumer Secret**:
+
+    ![Generate credentials](../../assets/img/apis/advanced-topics/external-identity-providers/generate-credentials.png){.cInlineImage-full}
+
+5. Click **Generate Credentials** to generate the **Consumer Key** and **Consumer Secret**.
+
+6. Click the **User Keys** tab.
+
+    ![User keys tab](../../assets/img/apis/advanced-topics/external-identity-providers/user-keys-tab.png){.cInlineImage-half}
+   
+
+    !!! tip
+        The **User Keys** tab is only enabled if you meet the following prerequisites:</br>
+          - Have at least one external IdP added to your organization.</br>
+          - Before you click **Generate Credentials**, click **Advanced Configurations** and make sure you select **Token Exchange** as a **Grant Type**.  
+
+           ![Application keys advanced configuration](../../assets/img/apis/advanced-topics/external-identity-providers/application-keys-advanced-configuration.png){.cInlineImage-half}
+    
+
+7. Select an IdP from the **Identity Provider** drop-down list.
+
+    ![Select an identity provider](../../assets/img/apis/advanced-topics/external-identity-providers/select-an-identity-provider.png){.cInlineImage-half}
+
+    This displays the steps you need to follow to generate a token:
+
+    ![User keys](../../assets/img/apis/advanced-topics/external-identity-providers/user-keys.png){.cInlineImage-half}
+
+8. As mentioned in **Step 1** in the UI, obtain an access token from the external IdP that you selected.
+   
+9. Paste the access token in the **<Your_IdP_name> Token** field. Here, you see the field name displayed as **Okta Token**.
+
+10. Click **://Curl** and copy the cURL command to obtain the Choreo access token. Alternatively, you can click **Generate Test Token** to get the Choreo access token via the UI.
```

Comment:
Our recommended way should be to generate through the UI right? shall we add that first? and mention cURL as an alternatiive?

Reply:
Done.